packages/vite/src/node/plugins/asset.ts

```diff
@@ -20,6 +21,8 @@ import { FS_PREFIX } from '../constants'
 
 export const assetUrlRE = /__VITE_ASSET__([a-z\d]{8})__(?:\$_(.*?)__)?/g
 
+export const duplicateAssets: OutputAsset[] = []
```

Comment:
We need this to be a `WeakMap` against the `ResolvedConfig` that gets reset on `buildStart` so `vite --watch` correctly gets a clean copy on each change. See how we keep other metadata to communicate between plugins here https://github.com/vitejs/vite/blob/4158b98c8fc30070967de1176f4b1c252d275386/packages/vite/src/node/plugins/asset.ts#L33

Reply:
Done! I've left it as an array value in a weak map against the config. Hope that does the trick!